### PR TITLE
Feature/fl

### PR DESCRIPTION
**Pull request by lzw9560**
FL add

### CODE REVIEW

**Review by Keepmoving-ZXY — APPROVED** [review]

File: build_local.sh [hunk 1]
```diff
@@ -32,7 +33,7 @@ fi
 BASE_DIR=`ls -l | grep bazel-bin | awk '{print $11}'`
 IMAGE_NAME="primihub/primihub_node"
 
-sed -i "10c ARG TARGET_PATH=$BASE_DIR" Dockerfile.local
+sed -i "12c ARG TARGET_PATH=$BASE_DIR" Dockerfile.local
```
Comment (yankaili2006) [inline build_local.sh:36]: sed -i 'XXc' 这种方式，在mac上有问题，替换一下用法。

Comment (yankaili2006) [reply build_local.sh:36]: sed -e "s|PLACEHOLDER-PYTHON3.X-CONFIG|${NEWLINE}|g" 参考

File: python/primihub/tests/xgb_guest.py [hunk 2]
```diff
@@ -22,19 +22,17 @@
...
+HOST_DATA_PATH = path.abspath(path.join(path.dirname(__file__), "data/wisconsin_host.data"))  # noqa
+GUEST_DATA_PATH = path.abspath(path.join(path.dirname(__file__), "data/wisconsin_guest.data"))  # noqa
 
 ph.context.Context.dataset_map = {
     'label_dataset': HOST_DATA_PATH,
     'guest_dataset': GUEST_DATA_PATH,
-    'test_dataset': TEST_DATA_PATH
 }
 
 ph.context.Context.output_path = "/home/zxy/primihub/python/primihub/tests/data/result/xgb_prediction.csv"
```
Comment (yankaili2006) [inline python/primihub/tests/xgb_guest.py:33]: change to relative path

File: build_local.sh (hunk 1 above)
Comment (yankaili2006) [inline build_local.sh:36]: sed 'XXc'  not compatible with mac, change to sed "s|PLACEHOLDER-PYTHON3.X-CONFIG|${NEWLINE}|g"

File: python/primihub/tests/xgb_guest.py (hunk 2 above)
Comment (lzw9560) [reply python/primihub/tests/xgb_guest.py:33]: @Keepmoving-ZXY

File: build_local.sh (hunk 1 above)
Comment (lzw9560) [reply build_local.sh:36]: @fuxingbit

File: python/primihub/tests/xgb_guest.py (hunk 2 above)
Comment (Keepmoving-ZXY) [reply python/primihub/tests/xgb_guest.py:33]: updated.

**Review by Keepmoving-ZXY — APPROVED** [review]

**Review by fuxingbit — APPROVED** [review]

**Review by yankaili2006 — APPROVED** [review]

**Review by hobo0cn — CHANGES_REQUESTED** [review]
pytest is a required item for each new feature.

**Review by hobo0cn — COMMENTED** [review]
@y981431999 pytest means analysis each test case,  not only warp main function...

**Review by hobo0cn — APPROVED** [review]

**Pull request merged**